src/main/groovy/edu/oregonstate/mist/inventory/resources/InventoryResource.groovy

```diff
@@ -89,22 +90,41 @@ class InventoryResource extends Resource {
         List<Error> errors = getErrors(newResultObject)
 
         if (errors) {
-            Response.ResponseBuilder responseBuilder = Response.status(Response.Status.BAD_REQUEST)
-            return responseBuilder.entity(errors).build()
+            return errorsResponse(errors)
         }
         Inventory newInventory = resultObjectToInventory(newResultObject)
         newInventory.id = newResultObject.data['id'] ?: randomUUID() as String
 
         inventoryDAOWrapper.createInventory(newInventory)
 
-        ResourceObject newInventoryFromDB = inventoryDAOWrapper.getInventoryById(newInventory.id)
+        ResourceObject inventoryFromDB = inventoryDAOWrapper.getInventoryById(newInventory.id)
 
-        if (!newInventoryFromDB) {
-            internalServerError(ErrorMessages.getInventoryError()).build()
+        if (!inventoryFromDB) {
+            getInventoryError()
         } else {
-            created(new ResultObject(
-                    data: newInventoryFromDB
-            )).build()
+            created(addSingleObjectToResultObject(inventoryFromDB)).build()
+        }
+    }
+
+    @Timed
+    @PUT
+    @Path('{id: [0-9a-zA-Z-]+}')
+    Response updateInventory(@PathParam('id') String inventoryID,
+                             @Valid ResultObject resultObject) {
+        List<Error> errors = getErrors(resultObject, inventoryID)
+
+        if (errors) {
+            return errorsResponse(errors)
+        }
+
+        inventoryDAOWrapper.updateInventory(resultObjectToInventory(resultObject), inventoryID)
+
+        ResourceObject inventoryFromDB = inventoryDAOWrapper.getInventoryById(inventoryID)
```

Comment:
Ok, fair enough. I think the exceptions being triggered and using the transaction should handle this scenario. Relying on the end user to parse through the response json in the payload and compare it to the submitted value though is not reliable (doesn't apply to this use case though).